infrastructure/storage-account.tf

```diff
@@ -0,0 +1,85 @@
+provider "azurerm" {
+  alias = "mgmt"
+  subscription_id = var.mgmt_subscription_id
+  skip_provider_registration = true
+  features {}
+}
+
+locals {
+  account_name = replace("${var.product}${var.env}", "-", "")
+  container_names = [
+    "new",
+    "backup",
+    "processed",
+    "zipped",
+    "encrypted"]
+  resourcegroup_name = azurerm_resource_group.rg.name
+
+  valid_subnets = [
+    data.azurerm_subnet.jenkins_subnet.id,
+    data.azurerm_subnet.jenkins_aks_00.id,
+    data.azurerm_subnet.jenkins_aks_01.id,
+    data.azurerm_subnet.app_aks_00_subnet.id,
+    data.azurerm_subnet.app_aks_01_subnet.id
+  ]
+
+  preview_subnets = var.env == "aat" ? [
+    data.azurerm_subnet.preview_aks_00_subnet.id,
+    data.azurerm_subnet.preview_aks_01_subnet.id] : []
+  all_valid_subnets = concat(local.valid_subnets, local.preview_subnets)
+
+}
+
+resource "azurerm_storage_account" "storage_account" {
+  name = local.account_name
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+  location = azurerm_resource_group.rg.location
+  account_tier = "Standard"
+  account_replication_type = "LRS"
+  account_kind = "BlobStorage"
```

Comment:
I think we better use the new version "General-purpose V2"

Reply:
replace account_kind = "StorageV2